ci/test_python.sh

```diff
@@ -159,7 +159,8 @@ if [[ "${RAPIDS_CUDA_VERSION}" == "11.8.0" ]]; then
       cugraph \
       cugraph-dgl \
       'dgl>=1.1.0.cu*,<=2.0.0.cu*' \
-      'pytorch>=2.0' \
+      'pytorch>=2.0,<2.4' \
+      'numpy>=2.0' \
```

Comment:
Are we able to relax the NumPy pin now?

Also may want to bump the lower bound on PyTorch here to ensure NumPy 2 compatibility, which was added to the conda-forge package in PR: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/238

There are a few more cases where this happens below, which we may want to give similar treatment

```suggestion
      'pytorch>=2.3,<2.4' \
```

Reply:
PyTorch 2.3+ is ok with me

Reply:
Thank you! I know you'd said that in https://github.com/rapidsai/cugraph/pull/4615#discussion_r1735256909 as well, this conversation is getting hard to follow split across multiple similar threads.

@hcho3 @jakirkham it seems to me like we have a clear path forward here:

* remove all the `numpy>2` that were temporarily added for testing
* set all `pytorch` / `torch` pins everywhere to `>=2.3,<2.4`
* bump `nccl` to `nccl>=2.19` (ref: https://github.com/rapidsai/build-planning/issues/102#issuecomment-2377268840)

Reply:
I just pulled this branch and searched like this:

```text
git grep -E 'orch.*2\.'
```

Looks like there's one more `pytorch=2.1.0` reference that probably should be updated to 2.3:

https://github.com/seberg/cugraph/blob/a6a84ebca91d0b0e8f1dabd0fa5f8be08283d82b/ci/test_python.sh#L198-L205

In a future PR, it'd be great to drive more of these things through `dependencies.yaml` and `rapids-dependency-file-generator`, to make it less likely that things like this are missed in upgrades.

Reply:
Yeah, I think this should be resolved in `cugraph-gnn`.  But if it isn't I agree we should fix it.  This got in there because of some of the quirks of the conda environment we use to test cuGraph.  `cugraph-gnn` aligns the package versions with PyTorch, and shouldn't have to do this.

Reply:
Done